### PR TITLE
Feature/nic actionplans

### PR DESCRIPTION
**Pull request by njjc**

### CODE REVIEW

File: src/classes/EP_EngagementPlans_TDTM.cls [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/*
```
Comment (kbromer) [inline src/classes/EP_EngagementPlans_TDTM.cls]: Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

Comment (kbromer) [inline src/classes/EP_EngagementPlans_TDTM.cls]: Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

File: src/classes/EP_EngagementPlans_TEST.cls [hunk 2]
```diff
@@ -0,0 +1,296 @@
+/*
```
Comment (kbromer) [inline src/classes/EP_EngagementPlans_TEST.cls]: * I suggest to annotate test methods with `@IsTest` for better readability.
* I suggest to annotate test methods with `@IsTest` for better readability.
* Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

Comment (kbromer) [inline src/classes/EP_EngagementPlans_TEST.cls]: * I suggest to annotate test methods with `@IsTest` for better readability.
* I suggest to annotate test methods with `@IsTest` for better readability.
* Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

File: src/classes/EP_EngagementPlans_UTIL.cls [hunk 3]
```diff
@@ -0,0 +1,124 @@
+/*
```
Comment (kbromer) [inline src/classes/EP_EngagementPlans_UTIL.cls]: Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

Comment (kbromer) [inline src/classes/EP_EngagementPlans_UTIL.cls]: Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

File: src/classes/EP_TaskDependency_TDTM.cls [hunk 4]
```diff
@@ -0,0 +1,209 @@
+/*
```
Comment (kbromer) [inline src/classes/EP_TaskDependency_TDTM.cls]: Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

Comment (kbromer) [inline src/classes/EP_TaskDependency_TDTM.cls]: Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

File: src/classes/EP_TaskDependency_TEST.cls [hunk 5]
```diff
@@ -0,0 +1,232 @@
+/*
```
Comment (kbromer) [inline src/classes/EP_TaskDependency_TEST.cls]: * I suggest to annotate test methods with `@IsTest` for better readability.
* Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

Comment (kbromer) [inline src/classes/EP_TaskDependency_TEST.cls]: * I suggest to annotate test methods with `@IsTest` for better readability.
* Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

File: src/classes/EP_Task_UTIL.cls [hunk 6]
```diff
@@ -0,0 +1,312 @@
+/*
```
Comment (kbromer) [inline src/classes/EP_Task_UTIL.cls]: * This method is empty, I suggest to remove it.
* Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

Comment (kbromer) [inline src/classes/EP_Task_UTIL.cls]: * This method is empty, I suggest to remove it.
* Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

File: src/triggers/TDTM_EngagementPlan.trigger [hunk 7]
```diff
@@ -0,0 +1,37 @@
+/*
```
Comment (kbromer) [inline src/triggers/TDTM_EngagementPlan.trigger]: Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

Comment (kbromer) [inline src/triggers/TDTM_EngagementPlan.trigger]: Classes and triggers should use the `UpperCamelCase` naming convention. I suggest to rename accordingly.

File: src/classes/UTIL_Namespace_TEST.cls [hunk 8]
```diff
@@ -105,7 +108,6 @@ public with sharing class UTIL_Namespace_TEST {
         str = Label.alloTotals;
         str = Label.alloCampaignExceedsOppAmount;
         str = Label.alloUnallocated;
-        str = Label.stgHelpBatchCreateDefault;
```
Comment (njjc) [inline src/classes/UTIL_Namespace_TEST.cls]: Undo this change.

File: src/layouts/Address__c-Address Layout.layout [hunk 9]
```diff
@@ -150,7 +150,7 @@
             <quickActionName>FeedItem.TextPost</quickActionName>
         </quickActionListItems>
         <quickActionListItems>
-            <quickActionName>Address__c.Update_Address</quickActionName>
+            <quickActionName>nictest__Address__c.Update_Address</quickActionName>
```
Comment (jjbennett) [inline src/layouts/Address__c-Address Layout.layout]: Reminder that this needs to be removed.

File: src/layouts/Contact-Contact Layout.layout [hunk 10]
```diff
@@ -508,6 +507,14 @@
         </relatedContentItems>
     </relatedContent>
     <relatedLists>
+        <fields>Engagement_Plan_Template__c</fields>
```
Comment (jjbennett) [inline src/layouts/Contact-Contact Layout.layout]: Reminder this should be removed as well.

File: src/layouts/Engagement_Plan_Task__c-Action Plan Task Layout.layout [hunk 11]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jjbennett) [inline src/layouts/Engagement_Plan_Task__c-Action Plan Task Layout.layout]: Reminder to remove this Action Plan Task Layout file.

File: src/layouts/Engagement_Plan__c-Action Plan Layout.layout [hunk 12]
```diff
@@ -0,0 +1,106 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jjbennett) [inline src/layouts/Engagement_Plan__c-Action Plan Layout.layout]: Reminder to remove this Action Plan Layout file.

File: src/classes/EP_Task_UTIL.cls [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+    /*******************************************************************************************************
+    * @description Creates a Task based on an Engagement Plan Task. Copies fields, sets up dependent tasks
+    * by providing a special status and determining delay in due date based on dependent tasks.
+    * @param epTask An Engagement Plan Task from which to copy fields.
+    * @return void This method updates class properties.
+    ********************************************************************************************************/
+    public Task createTask(Engagement_Plan_Task__c epTask, Engagement_Plan__c engagementPlan) {
+        Engagement_Plan_Template__c template = templateMap.get(epTask.Engagement_Plan_Template__c);
+
+        Map<String,String> taskAPTaskFieldMap = new Map<String,String>();
```
Comment (jjbennett) [inline src/classes/EP_Task_UTIL.cls]: Variable name still using AP vs EP.

**Comment by njjc** [comment]
@davidhabib this is good to go, feel free to schedule a time to review together.
